pkg/render/tiers/tiers.go

```diff
@@ -0,0 +1,333 @@
+// Copyright (c) 2022 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tiers
+
+import (
+	"strings"
+
+	"github.com/tigera/operator/pkg/render/kubecontrollers"
+
+	v3 "github.com/tigera/api/pkg/apis/projectcalico/v3"
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/render"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/render/common/networkpolicy"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	APIServerPolicyName      = networkpolicy.TigeraComponentPolicyPrefix + "cnx-apiserver-access"
+	ClusterDNSPolicyName     = networkpolicy.TigeraComponentPolicyPrefix + "cluster-dns"
+	KubeControllerPolicyName = networkpolicy.TigeraComponentPolicyPrefix + "kube-controller-access"
+	PacketCapturePolicyName  = networkpolicy.TigeraComponentPolicyPrefix + render.PacketCaptureName
+)
+
+var DNSIngressNamespaceSelector = createDNSIngressNamespaceSelector(
+	render.GuardianNamespace,
+	render.ComplianceNamespace,
+	render.DexNamespace,
+	render.ElasticsearchNamespace,
+	render.LogCollectorNamespace,
+	render.IntrusionDetectionNamespace,
+	render.KibanaNamespace,
+	render.ManagerNamespace,
+	common.TigeraPrometheusNamespace,
+	"tigera-skraper",
+	render.ECKOperatorNamespace,
+	render.PacketCaptureNamespace,
+	rmeta.APIServerNamespace(operatorv1.TigeraSecureEnterprise),
+	common.CalicoNamespace,
+)
+
+var defaultTierOrder = 100.0
+
+func Tiers(cfg *Config) render.Component {
+	return tiersComponent{cfg: cfg}
+}
+
+type Config struct {
+	Openshift      bool
+	ManagedCluster bool
+}
+
+type tiersComponent struct {
+	cfg *Config
+}
+
+func (t tiersComponent) ResolveImages(is *operatorv1.ImageSet) error {
+	return nil
+}
+
+func (t tiersComponent) Objects() ([]client.Object, []client.Object) {
+	objsToCreate := []client.Object{
+		t.allowTigeraTier(),
+		t.allowTigeraClusterDNSPolicy(),
+		t.allowTigeraAPIServerPolicy(),
+		t.allowTigeraKubeControllersPolicy(),
+		t.allowTigeraPacketCapturePolicy(),
+	}
+
+	// Delete equivalent policies under different namespaced names previously managed outside the operator.
```

Comment:
Not sure the best approach here, this probably needs to be changed.

There are three policies (kube-dns, kube-dns-egress, guardian-calico-cloud-egress) currently created during the cloud install process, and I've now brought them into the operator under different names.

We should determine how we'd like these existing policies to be handled on upgrade. The options I see are:

1. The operator is responsible for deleting them, and explicitly names them here
2. The updated cloud install script is responsible for deleting them
3. Do nothing in either place

Reply:
My preference would be (2) since it's less cruft in the operator accounting for resources the operator never owned itself. but not sure how feasible that is. CC @tmjd